javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/config/DeclarativeConfigEarlyInitAgentConfig.java

```diff
@@ -0,0 +1,121 @@
+ * io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties}) is initialized.
+ */
+public final class DeclarativeConfigEarlyInitAgentConfig implements EarlyInitAgentConfig {
+  private final OpenTelemetryConfigurationModel configurationModel;
+  private final ConfigProperties declarativeConfigProperties;
+  private final SdkConfigProvider configProvider;
+
+  DeclarativeConfigEarlyInitAgentConfig(String configurationFile) {
+    this.configurationModel = loadConfigurationModel(configurationFile);
+    configProvider = SdkConfigProvider.create(configurationModel);
+    this.declarativeConfigProperties =
+        new DeclarativeConfigPropertiesBridge(
+            configProvider, this.configurationModel.getLogLevel());
+  }
+
+  @Override
+  public boolean isAgentEnabled() {
+    return !Objects.equals(configurationModel.getDisabled(), true);
+  }
+
+  @Nullable
+  @Override
+  public String getString(String propertyName) {
+    return declarativeConfigProperties.getString(propertyName);
+  }
+
+  @Override
+  public boolean getBoolean(String propertyName, boolean defaultValue) {
+    return declarativeConfigProperties.getBoolean(propertyName, defaultValue);
+  }
+
+  @Override
+  public int getInt(String propertyName, int defaultValue) {
+    return declarativeConfigProperties.getInt(propertyName, defaultValue);
+  }
+
+  @Override
+  public void logEarlyConfigErrorsIfAny() {}
+
+  private static OpenTelemetryConfigurationModel loadConfigurationModel(String configurationFile) {
+    try (FileInputStream fis = new FileInputStream(configurationFile)) {
+      return DeclarativeConfiguration.parse(fis);
+    } catch (IOException e) {
+      throw new DeclarativeConfigException("unable to read " + configurationFile, e);
+    }
+  }
+
+  @Override
+  public AutoConfiguredOpenTelemetrySdk installOpenTelemetrySdk(ClassLoader extensionClassLoader) {
+    OpenTelemetrySdk sdk =
+        DeclarativeConfiguration.create(
+            this.configurationModel, SpiHelper.serviceComponentLoader(extensionClassLoader));
+    Runtime.getRuntime().addShutdownHook(new Thread(sdk::close));
+    GlobalOpenTelemetry.set(sdk);
+    GlobalConfigProvider.set(this.configProvider);
+
+    setForceFlush(sdk);
+
+    try {
+      Method method =
+          AutoConfiguredOpenTelemetrySdk.class.getDeclaredMethod(
+              "create",
+              OpenTelemetrySdk.class,
+              Resource.class,
+              ConfigProperties.class,
+              Object.class);
+      method.setAccessible(true);
+      return (AutoConfiguredOpenTelemetrySdk)
+          method.invoke(
+              null,
+              sdk,
+              Resource.getDefault(),
+              this.declarativeConfigProperties,
+              this.configProvider);
+    } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+      throw new IllegalStateException("Error calling create on AutoConfiguredOpenTelemetrySdk", e);
+    }
```

Comment:
instead of reflection you could add a class in the same package as the sdk classes and use that class to call the package private methods

Reply:
fixed